src/main/java/com/alvarium/annotators/PkiHttpAnnotator.java

```diff
@@ -0,0 +1,146 @@
+
+/*******************************************************************************
+ * Copyright 2022 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.nio.file.Path;
+
+import com.alvarium.annotators.http.ParseResult;
+import com.alvarium.annotators.http.ParseResultException;
+import com.alvarium.contracts.Annotation;
+import com.alvarium.contracts.AnnotationType;
+import com.alvarium.hash.HashType;
+import com.alvarium.sign.KeyInfo;
+import com.alvarium.sign.SignException;
+import com.alvarium.sign.SignProvider;
+import com.alvarium.sign.SignProviderFactory;
+import com.alvarium.sign.SignatureInfo;
+import com.alvarium.sign.SignType;
+import com.alvarium.utils.Encoder;
+import com.alvarium.utils.PropertyBag;
+
+import org.apache.http.client.methods.HttpUriRequest;
+
+class PkiHttpAnnotator extends AbstractAnnotator implements Annotator {
+  private final HashType hash;
+  private SignatureInfo signature;
+  private final AnnotationType kind;
+
+  protected PkiHttpAnnotator(HashType hash, SignatureInfo signature) {
+    this.hash = hash;
+    this.signature = signature;
+    this.kind = AnnotationType.PKIHttp;
+  }
+
+  public Annotation execute(PropertyBag ctx, byte[] data) throws AnnotatorException {
+    final String key = super.deriveHash(hash, data);
+
+    String host;
+    try {
+      host = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      throw new AnnotatorException("Cannot get host name", e);
+    }
+
+    HttpUriRequest request = ctx.getProperty(AnnotationType.PKIHttp.name(), HttpUriRequest.class);
```

Comment:
This method may throw an `IllegalArgumentException` . Wrap the statement around a `try/catch` block and throw the custom made `AnnotatorException` while passing the original exception as an argument to create an exception stack.